src/library_async.js

```diff
@@ -828,6 +889,155 @@ mergeInto(LibraryManager.library, {
     });
   },
 
+  /*
+   * Layout of an ASYNCIFY Fiber structure
+   *
+   *  0 Asyncify context (Asyncify.currData)
+   *  4 STACKTOP
+   *  8 STACK_MAX
+   * 12 STACK_BASE
+   */
+  $Fibers: {
+    continuations: {},  // fiber pointer -> JS continuation function mapping
+  },
+
+  emscripten_fiber_create__sig: 'iiiii',
+  emscripten_fiber_create__deps: ['$Fibers', 'malloc'],
+  emscripten_fiber_create: function(entryPoint, userData, stack, stackSize) {
+    var fiber = _malloc(16);
+
+    // init Asyncify context to null
+    {{{ makeSetValue('fiber', 0, 0, 'i32') }}};
+
+    // stack grows down on the WASM backend, so base is at the top
+    var stackBase = stack + stackSize;
+    var stackMax = stack;
+
+    // init stack pointers
+    {{{ makeSetValue('fiber',  4, 'stackBase', 'i32') }}};
+    {{{ makeSetValue('fiber',  8, 'stackMax',  'i32') }}};
+    {{{ makeSetValue('fiber', 12, 'stackBase', 'i32') }}};
+
+    // init continuation (call user-supplied function on first entry)
+    Fibers.continuations[fiber] = function() {
+#if STACK_OVERFLOW_CHECK
+      writeStackCookie();
+#endif
+      {{{ makeDynCall('vi') }}}(entryPoint, userData);
+    };
+
+    return fiber;
+  },
+
+  emscripten_fiber_create_from_current_context__sig: 'i',
+  emscripten_fiber_create_from_current_context__deps: ['$Fibers', 'malloc'],
+  emscripten_fiber_create_from_current_context: function() {
+    var fiber = _malloc(16);
+
+    {{{ makeSetValue('fiber',  0, 0, 'i32') }}};
+    {{{ makeSetValue('fiber',  8, 'STACK_MAX',  'i32') }}};
+    {{{ makeSetValue('fiber', 12, 'STACK_BASE', 'i32') }}};
```

Comment:
It seems like this is initialized the fiber to have the entire stack region?  This is why I suggest it be called  "get_main" or similar since its not actually getting the current fibre that is running.  We could also use "get_thread"?  or "get_primary"? 

Again I think I like the way windows does with "GetFiberData".. this could return current fibre.  I would create anything at all (unless its called for the first time and there is no running fibre, which case it would create the primordial fibre, just like above with STACK_MAX and STACK_BASE and return that).

I'm sorry it sounds like I'm being difficult, I do generally support this API!

Reply:
> It seems like this is initialized the fiber to have the entire stack region? This is why I suggest it be called "get_main" or similar since its not actually getting the current fibre that is running. We could also use "get_thread"? or "get_primary"?

`STACK_BASE` and `STACK_MAX` are overwritten when the fiber is switched, so it does actually get the current fiber's values.

https://github.com/emscripten-core/emscripten/blob/91e3c5f814a76a09052f7f7903daa86987b832a0/src/library_async.js#L1019

Reply:
> Again I think I like the way windows does with "GetFiberData".. this could return current fibre. I would create anything at all (unless its called for the first time and there is no running fibre, which case it would create the primordial fibre, just like above with STACK_MAX and STACK_BASE and return that).

I don't understand what you meant here. Windows' GetFiberData just returns the void pointer that was passed to CreateFiber, IIRC.

Reply:
Sorry I think I meant `GetCurrentFiber` .   The idea would be that you could always call that to get the current fibre (fibre id/handle) and that it wouldn't create one unless it was the first ever call.